machine_test.go

```diff
@@ -489,6 +498,12 @@ func testUpdateGuestDrive(ctx context.Context, t *testing.T, m *Machine) {
 	}
 }
 
+func testUpdateGuestNetworkInterface(ctx context.Context, t *testing.T, m *Machine) {
+	if err := m.UpdateGuestNetworkInterface(ctx, "1", models.PartialNetworkInterface{IfaceID: String("1")}); err != nil {
```

Comment:
This looks wrong. Not so sure why PartialNetworkInterface has iface_id, while the RESTful API's path also has iface_id.

Reply:
Yea, I think this shows we may want to revisit the design on how we expose this to users. We may want to rename the method to be something like `UpdateGuestNetworkRates` or something along those lines.
```go
func (m *Machine) UpdateGuestNetworkRates(ctx context.Context, ifaceID string, rates RateLimitSet, opts ...UpdateGuestNetworkRatesOpts) {
     // create partial network iface here with the iface id and rate limits
}
````

The reason why there is two is due to how Firecracker handles API requests. I remember opening a github issue about this with `PatchGuestDrive`.